src/main/host/thread_ptrace.c

```diff
@@ -600,29 +796,100 @@ SysCallCondition* threadptrace_resume(Thread* base) {
             case THREAD_PTRACE_CHILD_STATE_TRACE_ME:
                 debug("THREAD_PTRACE_CHILD_STATE_TRACE_ME");
                 break;
+            case THREAD_PTRACE_CHILD_STATE_IPC_SYSCALL: {
+                debug("THREAD_PTRACE_CHILD_STATE_IPC_SYSCALL");
+                SysCallReturn ret = syscallhandler_make_syscall(thread->sys, &thread->syscall_args);
+                switch (ret.state) {
+                    case SYSCALL_BLOCK:
+                        debug("ipc_syscall blocked");
+                        // Don't leave it spinning.
+                        _threadptrace_ensureStopped(thread);
+                        return ret.cond;
+                    case SYSCALL_DONE: {
+                        debug("ipc_syscall done");
+                        ShimEvent shim_result = {
+                        .event_id = SHD_SHIM_EVENT_SYSCALL_COMPLETE,
+                        .event_data = {
+                            .syscall_complete = {.retval = ret.retval,
+                                                 .simulation_nanos =
+                                                     worker_getEmulatedTime()},
+
+                        }};
+                        shimevent_sendEventToPlugin(_threadptrace_ipcData(thread), &shim_result);
+                        break;
+                                       }
```

Comment:
Is this bracket is in a weird location? I expected it to be one tab back from the tab-stop on the previous line, but maybe this is where clang-format placed it?